docs/architecture/app-management.md

```diff
@@ -0,0 +1,315 @@
+# Automatic Application Management
```

Comment:
hmm, maybe it would be better to change file name - `app-management.md` is a bit too generic. What about `auto-app-management.md` or something like that?